src/main/java/com/google/step/SaveImageCloud.java

```
@@ -59,7 +61,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 byte[] imageData = getImageData(requestUrls.get(i));
                 saveImageToCloudStorage(imageData, mapImages.get(i));
             } catch (DeadlineExceededException e) {
-                // TODO: add error handling
+                LOGGER.log(Level.SEVERE, e.getMessage());
```

Probably this isn't how you'd want to handle this error, right?  Adding logging is good - much better than silently swallowing the error - but you'll probably want to do some sorta retry here?  Or perhaps, if a retry isn't going to happen, you're going to want to fire an alert or something so we notice that the upload failed?

In the worst case, you'd at least want to let the exception bubble out of this method so we get a crash when this fails.

---

The retry is handled by cron, which called this servlet originally. However, the retry only happens on a non-200 status returned from the server, which I don't think happens in it's current state. I figure that the sever log could act as an alert that the upload failed, but in regards to having the exception bubble out of this method: 

How do I simultaneously log an error and have the exception bubble out and crash properly so cron can perform a retry. Right now I'm thinking
`   // There are an equal number of elements in mapImages & requestUrls.
    function foo() throws DeadlineExceededException {
        for (int i = 0; i < mapImages.size(); i++) {
            try {
                byte[] imageData = getImageData(requestUrls.get(i));
                saveImageToCloudStorage(imageData, mapImages.get(i));
            } catch (DeadlineExceededException e) {
                LOGGER.log(Level.SEVERE, e.getMessage());
                throw DeadlineExceededException;
        }
    }`
but I don't think this is the best practice for throwing an error

---

I think you could just say "throw e;" at the end there, rather than throwing a new DeadlineExceededException.